modules/oe_webtools_etrans/src/Plugin/Block/UnifiedEtransBlock.php

```diff
@@ -306,4 +311,27 @@ protected function preparesWtEtransJson(string $translation_from, string $placeh
     return $json;
   }
 
+  /**
+   * Attempt to map langcode to ISO 639-1.
+   *
+   * @param string $langcode
+   *   The Drupal langcode.
+   *
+   * @return string
+   *   The ISO 639-1 langcode format.
+   */
+  protected function mapLangcodeToIso(string $langcode): string {
+    // In Drupal, langcode are often in ISO 639-1 format by default.
+    // But, some languages use browser formats like 'pt-pt' or 'ta-lk'.
+    // A mapping exists in the language module.
+    $mapping = array_flip(language_get_browser_drupal_langcode_mappings());
```

Comment:
One problem with this function is that it does not support dependency injection.
It might be cleaner to just use the config.
The only thing this function does differently is checking for `$config->isNew()`. The function was added in https://www.drupal.org/node/365615 in 2012. In this commit it was also used in a form, and this is probably the reason why they call `$config->isNew()`.
For us there is probably no reason why we would need this check. So it would be ok if we simply use the config.
The only other place where this is called in core is in ckeditor5.